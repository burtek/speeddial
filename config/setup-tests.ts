import '@testing-library/jest-dom';
import { createSerializer } from '@emotion/jest';
import i18n from 'i18next';
import 'jest-extended';
import 'jest-extended/all';
import { initReactI18next } from 'react-i18next';


expect.addSnapshotSerializer(createSerializer());

await i18n
    .use(initReactI18next)
    .init({
        lng: 'dev',
        interpolation: { escapeValue: false }
    });
