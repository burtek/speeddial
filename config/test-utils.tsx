import type { RenderOptions } from '@testing-library/react';
import { render } from '@testing-library/react';
import type { FC, PropsWithChildren, ReactElement } from 'react';
import { Provider as StoreProvider } from 'react-redux';

import { store } from '@@data/index';
import { ThemedApp } from 'src/theme/themed-app';

const AllTheProviders: FC<PropsWithChildren> = ({ children }) => (
    <StoreProvider store={store}>
        <ThemedApp>
            {children}
        </ThemedApp>
    </StoreProvider>
);

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
// eslint-disable-next-line import/export
export * from '@testing-library/react';

// override render method
// eslint-disable-next-line import/export
export { customRender as render };
