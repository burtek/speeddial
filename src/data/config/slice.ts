import type { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from '@reduxjs/toolkit';

export type ThemeMode = 'dark' | 'light';

export const initialState = () => ({
    modalOpen: false,
    themeMode: null as ThemeMode | null
});

export const { actions, name: sliceName, reducer } = createSlice({
    name: 'config',
    initialState,
    reducers: {
        openModal(state) {
            state.modalOpen = true;
        },
        closeModal(state) {
            state.modalOpen = false;
        },

        setThemeMode(state, { payload }: PayloadAction<{ mode: ThemeMode | null }>) {
            state.themeMode = payload.mode;
        }
    }
});
