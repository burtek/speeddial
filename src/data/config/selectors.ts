import { createSelector } from '@reduxjs/toolkit';

import type { RootState } from '..';

import { sliceName } from './slice';


export const getConfig = (state: RootState) => state[sliceName];

export const getThemeMode = createSelector(getConfig, s => s.themeMode);

export const getIsDialogOpen = createSelector(getConfig, s => s.modalOpen);
