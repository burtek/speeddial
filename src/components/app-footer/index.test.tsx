import { render } from 'config/test-utils';

import { AppFooter } from './index';


describe('app-footer', () => {
    beforeAll(() => {
        vitest.spyOn(Date.prototype, 'getFullYear').mockReturnValue(2021);
    })
    
    afterAll(() => {
        vitest.mocked(Date.prototype.getFullYear).mockRestore();
    })

    it('should match snapshot', () => {
        const { container } = render(<AppFooter />);

        expect(container).toMatchSnapshot();
    });
});
