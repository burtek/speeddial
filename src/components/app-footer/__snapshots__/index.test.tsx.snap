// Vitest Snapshot v1

exports[`app-footer > should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-xi606m"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Â©
       
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="https://github.com/burtek"
        referrerpolicy="no-referrer"
        rel="noopener noreferrer nofollow"
        target="_blank"
      >
        burtek
      </a>
       2021 | 
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="https://github.com/burtek/speeddial"
        referrerpolicy="no-referrer"
        rel="noopener noreferrer nofollow"
        target="_blank"
      >
        footer.openSource
      </a>
       | footer.build test
    </p>
  </div>
</div>
`;
